File: LeetCode/1-100/28. 实现 strStr().js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  if (!needle) return 0
  let n = needle.length
  for (let i = 0; i <= haystack.length - n; ++i) {
    let str = haystack.slice(i, i + n)
    if (str === needle) {
      return i
    }
  }
  return -1
}

File: LeetCode/1-100/28. 实现 strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
    int len = haystack.length();
        int len2 = needle.length();
        for (int i = 0; i < len - len2 + 1; i++) {
            if (haystack.substring(i,i+ len2).equals(needle)) return i;
        }
        return -1;
    }
}